include/picongpu/plugins/radiation/Radiation.hpp

```diff
@@ -664,396 +659,197 @@ namespace picongpu
-
-                    std::string basePath("/data/%T/");
-                    splash::ColTypeString ctBasePath(basePath.length());
-                    hdf5DataFile.writeGlobalAttribute(ctBasePath, "basePath", basePath.c_str());
-
-                    splash::ColTypeString ctMeshesPath(meshesPathName.length());
-                    hdf5DataFile.writeGlobalAttribute(ctMeshesPath, "meshesPath", meshesPathName.c_str());
-
-
-                    splash::ColTypeString ctParticlesPath(particlesPathName.length());
-                    hdf5DataFile.writeGlobalAttribute(ctParticlesPath, "particlesPath", particlesPathName.c_str());
 
-                    std::string iterationEncoding("fileBased");
-                    splash::ColTypeString ctIterationEncoding(iterationEncoding.length());
-                    hdf5DataFile.writeGlobalAttribute(
-                        ctIterationEncoding,
-                        "iterationEncoding",
-                        iterationEncoding.c_str());
-
-                    /* the ..._0_0_0... extension comes from the current filename
-                       formating of the serial data colector in libSplash */
-                    const int indexCutDirectory = name.rfind('/');
-                    std::string iterationFormat(name.substr(indexCutDirectory + 1) + std::string("%T_0_0_0.h5"));
-                    splash::ColTypeString ctIterationFormat(iterationFormat.length());
-                    hdf5DataFile.writeGlobalAttribute(ctIterationFormat, "iterationFormat", iterationFormat.c_str());
-
-                    hdf5DataFile.writeAttribute(currentStep, splashFloatXType, nullptr, "dt", &DELTA_T);
+                    openPMDdataFileIteration.setDt<float>(DELTA_T);
                     const float_X time = float_X(currentStep) * DELTA_T;
-                    hdf5DataFile.writeAttribute(currentStep, splashFloatXType, nullptr, "time", &time);
-                    splash::ColTypeDouble ctDouble;
-                    hdf5DataFile.writeAttribute(currentStep, ctDouble, nullptr, "timeUnitSI", &UNIT_TIME);
-
+                    openPMDdataFileIteration.setTime(time);
+                    openPMDdataFileIteration.setTimeUnitSI(UNIT_TIME);
                     /* end required openPMD global attributes */
 
                     /* begin recommended openPMD global attributes */
-
-                    std::string author = Environment<>::get().SimulationDescription().getAuthor();
-                    if(author.length() > 0)
-                    {
-                        splash::ColTypeString ctAuthor(author.length());
-                        hdf5DataFile.writeGlobalAttribute(ctAuthor, "author", author.c_str());
-                    }
-
-                    std::string software("PIConGPU");
-                    splash::ColTypeString ctSoftware(software.length());
-                    hdf5DataFile.writeGlobalAttribute(ctSoftware, "software", software.c_str());
-
+                    const std::string software("PIConGPU");
```

Comment:
Note: `picongpu::openpmd::WriteMeta` in `include/picongpu/plugins/common/openPMDWriteMeta.hpp` might also be useful here. But not necessary for this PR, unifying the metadata writing across openPMD plugins is something that we'll have to do anyway.

Reply:
Thanks, I will keep this in mind.